### PR TITLE
Adding documentation to CWE-595 as part of #531

### PR DESCRIPTION
**Pull request by s19110**
Adding documentation to CWE-595 as part of #531

### CODE REVIEW

File: docs/Secure-Coding-Guide-for-Python/CWE-697/CWE-595/README.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
+# CWE-595: Comparison of Object References Instead of Object Contents
+
+In Python, the == operator is implemented by the `__eq__` method on an object [[python.org data model 2023](https://docs.python.org/3/reference/datamodel.html?highlight=__eq__#object.__eq__)]. For built-in types like `int` and `str`, the comparison is implemented in the interpreter. The main issue comes when implementing custom classes, where the default implementation compares object references using the `is` operator. The `is` operator compares the identities of the objects, equivalent to `id(obj1) == id(obj2)`. The `id` function is built into Python, and in the CPython interpreter, the standard implementation, it returns the object's memory address [[de Langen 2023](https://realpython.com/python-is-identity-vs-equality/)].
```
Comment (myteron) [inline docs/Secure-Coding-Guide-for-Python/CWE-697/CWE-595/README.md:3]: surround == with ticks `==`

File: docs/Secure-Coding-Guide-for-Python/CWE-697/CWE-595/README.md [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+You want to implement the `__eq__` method on a class if you believe you ever want to compare it to another object or find it in a list of objects. Actually, it is so common that the `dataclasses.dataclass` decorator by default implements it for you [[dataclasses — Data Classes — Python 3.11.4 documentation](https://docs.python.org/3/library/dataclasses.html#dataclasses.dataclass)].
+
+## Non-Compliant Code Example
+
+The non-compliant code shows how the default comparison operator compares object references rather than the object values. Furthermore, it displays how this causes issues when comparing lists of objects, although it applies to other types of collections as well. Finally, it shows how the `in` operator also depends on the behavior of the `__eq__` method and, therefore, also returns a non-desirable result.
```
Comment (myteron) [inline docs/Secure-Coding-Guide-for-Python/CWE-697/CWE-595/README.md:9]: The `noncompliant01.py`

further down typo `in` should be `int`

File: docs/Secure-Coding-Guide-for-Python/CWE-697/CWE-595/README.md [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+[*noncompliant01.py:*](noncompliant01.py)
```
Comment (myteron) [inline docs/Secure-Coding-Guide-for-Python/CWE-697/CWE-595/README.md:11]: add the following to all source code files;
```
# SPDX-FileCopyrightText: OpenSSF project contributors
# SPDX-License-Identifier: MIT
```
Only needs to be in the files, not in the readme.md.

**Review by myteron — COMMENTED** [review]
only minor changes

File: docs/Secure-Coding-Guide-for-Python/CWE-697/CWE-595/README.md (hunk 2 above)
Comment (s19110) [reply docs/Secure-Coding-Guide-for-Python/CWE-697/CWE-595/README.md:9]: The `in` operator is referencing this print:
print(Integer(12) in [Integer(10), Integer(12)])
I think this is not a typo in this case.

File: docs/Secure-Coding-Guide-for-Python/CWE-697/CWE-595/README.md (hunk 1 above)
Comment (s19110) [reply docs/Secure-Coding-Guide-for-Python/CWE-697/CWE-595/README.md]: Done

File: docs/Secure-Coding-Guide-for-Python/CWE-697/CWE-595/README.md (hunk 3 above)
Comment (s19110) [reply docs/Secure-Coding-Guide-for-Python/CWE-697/CWE-595/README.md:11]: Done

**Review by gkunz — APPROVED** [review]
LGTM

**Comment by myteron** [comment]
+1

**Pull request merged**